src/Data/ProgrammesDb/EntityRepository/CoreEntityRepository.php

```diff
@@ -186,13 +188,17 @@ public function findByPidFull(string $pid, string $entityType = 'CoreEntity'): ?
             ->setParameter('pid', $pid);
 
         $result = $qb->getQuery()->getOneOrNullResult(Query::HYDRATE_ARRAY);
-        $withHydratedParents = $result ? $this->resolveParents([$result])[0] : $result;
-        return $withHydratedParents ? $this->resolveCategories([$withHydratedParents])[0] : $withHydratedParents;
+        if (!$result) {
+            return $result;
+        }
+        $this->addToAncestryCache([$result]);
```

Comment:
I'm guessing this is in findByPid full, because resolveParents does not store the current base programme in the cache (only its ancestry), and on things like the tlec page we are getting lots of things that are immediate children of this top level item, thus we can sidestep future ancestry requests for them by having this parent already stored.

Reply:
You are technically correct. The best kind of correct.